daffodil-cli/src/it/scala/org/apache/daffodil/debugger/TestCLIDebugger.scala

```diff
@@ -155,21 +155,21 @@ class TestCLIdebugger {
       shell.sendLine(cmd)
       shell.expect(contains("(debug)"))
 
-      shell.sendLine("display eval (.)")
+      shell.sendLine("display eval (dfdl:occursIndex())")
       shell.sendLine("step")
-      shell.expect(contains("matrix"))
+      shell.expect(contains("1"))
 
       shell.sendLine("info displays")
-      shell.expect(contains("1: eval (.)"))
```

Comment:
So if we evaluate "." in the debugger it throws? We should catch this and output something about having no value.

Reply:
There's a place in the debugger with a big bunch of catches for this sort of thing, e.g., you type an expression at the debugger which divides by zero. That should get caught similarly, and not throw you out of the debugger.

Reply:
Yes it does. That's what was breaking the IT. 
Cool, I'll take a look.

Reply:
I suspect thismight get fixed by change it to call ``doSDE()`` rather than throwing the new Exception. I think the debugger explicitly handles SDE's (and also InfosetNoDataException) as special. So making this an SDE might fix it.

It might also make sense to have this new exception extend InfosetNoDataException, but with a new error message. That way things that expected an InfosetNoDataException will still work as before, but will have a more appropriate error message.